### DISCUSSION TITLE
How to get NAT traffic and subnet of wsl2?

### DISCUSSION THREAD

**Discussion by xiangnanscu** [microsoft/WSL#5483] [General]
Hello, I'm trying to use sshuttle in windows 10 via wsl2. I come from this issue https://github.com/sshuttle/sshuttle/blob/master/docs/windows.rst. 

Now I want to know how to get WSL2's equivalent ips of `10.0.0.0/8`, `192.168.0.0/16` and `192.168.1.200` referenced in Vagrant.
Is it possible?

**the origin text**
Microsoft Windows
=================
Currently there is no built in support for running sshuttle directly on
Microsoft Windows.

What we can really do is to create a Linux VM with Vagrant (or simply
Virtualbox if you like). In the Vagrant settings, remember to turn on bridged
NIC. Then, run sshuttle inside the VM like below::

    sshuttle -l 0.0.0.0 -x 10.0.0.0/8 -x 192.168.0.0/16 0/0

10.0.0.0/8 excludes NAT traffic of Vagrant and 192.168.0.0/16 excludes
traffic to local area network (assuming that we're using 192.168.0.0 subnet).

Assuming the VM has the IP 192.168.1.200 obtained on the bridge NIC (we can
configure that in Vagrant), we can then ask Windows to route all its traffic
via the VM by running the following in cmd.exe with admin right::

     route add 0.0.0.0 mask 0.0.0.0 192.168.1.200

**Comment by therealkenc** [comment]
The link cited in the OP states:

> What we can really do is to create a Linux VM with Vagrant (or simply Virtualbox if you like). In the Vagrant settings, remember to turn on bridged NIC. Then, run sshuttle inside the VM like below:

That generally won't bode well on WSL2 because #4150 et al, if you take the above bridged networking requirement (contrast NAT) at face value.

A brief look at `sshuttle` makes me wonder if it might be possible to get working though. There is no bug or actionable here, but the discussion can continue in the newly minted discussion forum.
